lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -54,27 +59,48 @@ public TaskServiceImpl(TaskanaEngine taskanaEngine, TaskMapper taskMapper,
         this.taskanaEngineImpl = (TaskanaEngineImpl) taskanaEngine;
         this.taskMapper = taskMapper;
         this.objectReferenceMapper = objectReferenceMapper;
+        this.workbasketService =  taskanaEngineImpl.getWorkbasketService();
     }
 
     @Override
-    public Task claim(String id, String userName) throws TaskNotFoundException {
+    public Task claim(String id)  throws TaskNotFoundException, InvalidStateException, InvalidOwnerException {
```

Comment:
Why do we need two different claim() Methods? And maybe it should be called claimTask().